### PR TITLE
vqe code update - primitives, qiskit-nature upgrade (0.5.2)

### LINKED ISSUE #1840: Update code in VQE page
### Feature description

We should update some of the code in the VQE page in advance of upcoming deprecation:

- [ ] Import from `qiskit_aer` instead of `qiskit.providers`
- [ ] ~~Migrate to Qiskit Nature `5.0.0` ([migration guide](https://qiskit.org/documentation/nature/migration/index.html))~~ Moved to #1893
- [ ] Use QubitConvertor/ParityMapper instead of opflow
- [ ] Use estimator primitive ([reference tutorial](https://qiskit.org/documentation/tutorials.html#algorithms))

### Additional context

Thanks to @woodsp-ibm (see comments in https://github.com/Qiskit/platypus/pull/1820 thread).

### PR DESCRIPTION
**Pull request by david-alber**
## Changes

Updates the vqe-molecule.ipynb example with the primitives Sampler and Estimator from qiskit_aer. 
The QubitConvertor/ParityMapper from qiskit_nature.second_q (latest advancements) are used.
The update required to upgrade qiskit-nature to 0.5.2.

Closes #1840

### CODE REVIEW

**Comment by david-alber** [comment]
Pylint PR issues of trailing whitespaces and qiskit.primitives.are import are resolved with commit fb5deb54cf0b759a46eaec431014b363de6bff11

**Review by frankharkins — CHANGES_REQUESTED** [review]
Great work, thank you! I just have a couple of small comments.

File: notebooks/ch-applications/vqe-molecules.ipynb [hunk 1]
```diff
@@ -592,16 +596,18 @@
...
-    "from qiskit.providers.fake_provider import FakeManila\n",
-    "from qiskit.providers.aer.noise import NoiseModel # pylint: disable=import-error, no-name-in-module\"\n",
-    "backend = Aer.get_backend('aer_simulator')\n",
-    "device = FakeManila()\n",
-    "NOISE_MODEL = NoiseModel.from_backend(device)\n",
-    "coupling_map = device.configuration().coupling_map\n",
-    "qi = qiskit.utils.QuantumInstance(backend,\n",
-    "                    coupling_map=coupling_map,\n",
-    "                    noise_model=NOISE_MODEL,\n",
-    "                    measurement_error_mitigation_cls=qiskit.utils.mitigation.CompleteMeasFitter)"
```
[the remark this answers is not in the document]
Comment (david-alber) [reply notebooks/ch-applications/vqe-molecules.ipynb:604]: Tackling error mitigation via the `resilinece_level` option is (as for now) not possiblie with the AerSimulator, see [here](https://github.com/Qiskit/qiskit-aer/issues/1658).
However, [IBM Runtime qasm simulator] (https://qiskit.org/documentation/partners/qiskit_ibm_runtime/how_to/noisy_simulators.html#set-the-backend-and-options-to-use) could be used. This requires to use the qiskit-ibm-runtime module.

[the remark this answers is not in the document]
Comment (david-alber) [reply notebooks/ch-applications/vqe-molecules.ipynb:604]: I will add a note at the end of the notebook ;)

File: notebooks/ch-applications/vqe-molecules.ipynb [hunk 2]
```diff
@@ -395,39 +394,41 @@
...
+    "When using an actual quantum device, or a true simulator such as the `AerSimulator` with `VQE(estimator, var_form, optimizer)` the calculation of the expectation value is as follows: A Hamiltonian is be represented as a sum of a Pauli strings, with each Pauli term acting on a qubit as specified by the mapping being used. Each Pauli string has a corresponding circuit appended to the circuit corresponding to $|\\psi(\\theta)\\rangle$. Subsequently, each of these circuits is executed, and all of the results are used to determine the expectation value of $H$ on $|\\psi(\\theta)\\rangle$. \n",
```
Comment (frankharkins) [inline notebooks/ch-applications/vqe-molecules.ipynb:431]:
```suggestion
    "When using an actual quantum device, or a true simulator such as the `AerSimulator` with `VQE(estimator, var_form, optimizer)` the calculation of the expectation value is as follows: A Hamiltonian is represented as a sum of a Pauli strings, with each Pauli term acting on a qubit as specified by the mapping being used. Each Pauli string has a corresponding circuit appended to the circuit corresponding to $|\\psi(\\theta)\\rangle$. Subsequently, each of these circuits is executed, and all of the results are used to determine the expectation value of $H$ on $|\\psi(\\theta)\\rangle$. \n",
```

**Review by frankharkins — APPROVED** [review]
Looks good! Thanks again.

**Pull request merged**